src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -0,0 +1,327 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers.FrameworkEventSource;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    public enum PipeMode
+    {
+        Logs = 1,
+        Metrics,
+    }
+
+    public class DiagnosticsEventPipeProcessor : IAsyncDisposable
```

Comment:
@noahfalk Per our discussion, this would likely encapsulate the Pipeline boundary for EventPipe. Currently this class, along with DiagnosticsMonitor, should be able to handle all the EventPipe scenarios.